sdk/openai/openai/test/public/assistants.spec.ts

```diff
@@ -0,0 +1,397 @@
+            type: "function" as const,
+            function: {
+              name: favoriteCityFunctionName,
+              description: favoriteCityFunctionDescription,
+              parameters: {
+                type: "object",
+                properties: {},
+              },
+            },
+          };
+
+          const getCityNickname = (city: string): string => {
+            switch (city) {
+              case "Atlanta, GA":
+                return "The ATL";
+              case "Seattle, WA":
+                return "The Emerald City";
+              case "Los Angeles, CA":
+                return "LA";
+              default:
+                return "Unknown";
+            }
+          };
+
+          const getCityNicknameFunctionName = "getCityNickname";
+          const getCityNicknameFunctionDescription =
+            "Gets the nickname for a city, e.g. 'LA' for 'Los Angeles, CA'.";
+          const getCityNicknameTool = {
+            type: "function" as const,
+            function: {
+              name: getCityNicknameFunctionName,
+              description: getCityNicknameFunctionDescription,
+              parameters: {
+                type: "object",
+                properties: {
+                  city: {
+                    type: "string",
+                    description: "The city and state, e.g. San Francisco, CA",
+                  },
+                },
+              },
+            },
+          };
+
+          let favoriteCityCalled = false;
+          let nicknameCalled = false;
+          const getResolvedToolOutput = (toolCall: {
+            id: string;
+            function?: any;
+          }): { output: string } => {
+            const toolOutput = { tool_call_id: toolCall.id, output: "" };
+            if (toolCall["function"]) {
+              const functionCall = toolCall["function"];
+              const functionName = functionCall.name;
+              const functionArgs = JSON.parse(functionCall["arguments"] ?? {});
+              switch (functionName) {
+                case favoriteCityFunctionName:
+                  toolOutput.output = getFavoriteCity();
+                  favoriteCityCalled = true;
+                  break;
+                case getCityNicknameFunctionName:
+                  toolOutput.output = getCityNickname(functionArgs["city"]);
+                  nicknameCalled = true;
+                  break;
+                default:
+                  toolOutput.output = `Unknown function: ${functionName}`;
+                  break;
+              }
+            }
+
+            return toolOutput;
+          };
+
+          const instructions = `You are a helpful assistant. Use the provided functions to help answer questions.
+            Customize your responses to the user's preferences as much as possible and use friendly
+            nicknames for cities whenever possible.
+        `;
+          const functionAssistant = {
+            model: "gpt-4-1106-preview",
```

Comment:
could we factor out the deployment name from inside the tests into one central place for now so we can make it easier to run these tests with several deployments later?